### DISCUSSION TITLE
Display Reality Model mesh

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#2060] [Q&A] [answered]
Is it possible to display the mesh underlying the Reality model.  Please advise.

Apparently Bentley's Acute3d viewer has this option which allows much more precision in selecting points for measuring tools (a request from ex Acute3d viewer user)

Thanks in advance for you guidance on this request.

Ovais

**Comment by pmconne** [comment]
Please be more specific about how you want to visualize the underlying mesh.
We could draw it without the textures quite easily, but it would not be lit.
Or are you asking for wiremesh display?

**Reply by ovaisa** [reply]
Hi Paul,

I believe it should be OK if the mesh is not lit.  Let's start with simple solution, if you can guide me how to draw reality model without textures, I can try it, hopefully it will be sufficient.

Ovais

**Reply by pmconne** [reply]
It will require code changes on our part.

**Reply by ovaisa** [reply]
Hi Paul,

In your earlier response, you stated "We could draw it without the textures quite easily", does it imply that it can be done easily in the iTwin platform but is not possible using the existing API.

If so, I would like to request to add this capability to the iTwin backlog

I am attaching an image if our model with mesh turned on (from Descartes) to show what I am looking for.  This image shows the underlying textures but this is not our requirement.  Our need is more utilitarian, to measure distances more accurately, visualization is not a consideration.  If the model is all monochrome and not lit, this should be sufficient for what our users are looking for.

![Mesh](https://user-images.githubusercontent.com/83007711/129451441-9c17a87d-3b5e-45fd-a657-5027b59c71f1.JPG)

**Reply by pmconne** [reply]
Thanks. An unlit monochrome mesh would not make precise measurement any easier. Based on your image you are asking for wiremesh display. Added to the backlog.

**Reply by ovaisa** [reply]
Thank you for adding our request to the backlog.  I am attaching an image from Acute3D that our user has passed on to me to show how he is using the mesh to snap to exact points they want to use for measurements.

Currently it seems that trying to snap to reality mesh is somewhat unpredictable, sometime the snapped point is nowhere close (in 3D space) to the point intended to be selected.  Is there a some setting we can turn on that will ensure that selected point is snapped to a mesh point.

Measurements is a very basic and important capability for our user, would appreciate if our request can be given priority.

Thank you,

Ovais

![Mesh](https://user-images.githubusercontent.com/83007711/129752724-c4436a12-2927-406d-b28c-921d06c2cf76.png)

**Reply by pmconne** [reply]
Just to be clear, you are requesting **two** new features.
1. Wiremesh display.
2. Snapping to vertices of reality mesh.

There is currently no setting that will help you measure more precisely. The geometry of the reality mesh is not accessible to us on the CPU. We obtain the meshes chopped up into 3d tiles optimized for transfer to the GPU. Perhaps @RBBentley has some estimate of how tricky it would be to snap to those tiles if we retained them on the CPU.

**Reply by ovaisa** [reply]
Hi Paul,

Our main requirement is ability to measurement more accurately.  As I mentioned, our user is unable to select right location, sometime it is off by a long.  I was hoping that with the wire mesh on, it will give them visual aid to select a point on mesh which is close to point where they want to measure.   If we are able to turn on the wire mesh and it is able to achieve this goal, that is all we would need.  My question on snapping was more for the interim until wire mesh option becomes available.

**Comment by ovaisa** [comment]
Hi, Just checking on when we can expect the capability of showing the underlying mesh would be available.  An update would be much appreciated....

Thank you,

**Reply by pmconne** [reply]
We have not yet scheduled this feature.

**Comment by pmconne** [comment, marked as answer]
@ovaisa this is implemented in #2787.
